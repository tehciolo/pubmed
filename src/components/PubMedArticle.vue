<template>
  <article class="card">
    <header class="card-header">
      <p class="card-header-title">{{ article.title }}</p>
    </header>
    <div class="card-content">
      <p class="content">
        <strong>Publication date:</strong> {{ article.pubdate }}<br />
        <strong>Journal:</strong> {{ article.fulljournalname }}<br />
        <strong>Author(s):</strong> {{ authors }}<br />
      </p>
    </div>
    <footer class="card-footer">
      <p class="card-footer-item">
        <span> View on <a :href="href">PubMed</a> </span>
      </p>
    </footer>
  </article>
</template>

<script lang="ts">
import { defineComponent } from "@vue/composition-api";

export default defineComponent({
  name: "PubMedArticle",

  props: {
    article: {
      type: Object,
      required: true,
    },
  },

  computed: {
    authors() {
      return this.article.authors.map(({ name }) => name).join(", ");
    },

    href() {
      return `https://pubmed.ncbi.nlm.nih.gov/${this.article.uid}/`;
    },
  },
});
</script>
