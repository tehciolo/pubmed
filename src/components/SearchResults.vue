<template>
  <section class="section is-medium">
    <p v-if="isPristine">
      PubMed is a free resource supporting the search and retrieval of biomedical and life
      sciences literature with the aim of improving health â€“ both globally and personally.
      Enter a search term above to start.
    </p>
    <div v-else>
      <p class="subtitle">We have found {{ count }} article(s).</p>

      <ul>
        <li v-for="result in results" :key="result.uid">
          <PubMedArticle :article="result" />
        </li>
      </ul>
    </div>
  </section>
</template>

<script lang="ts">
import { defineComponent } from "@vue/composition-api";
import PubMedArticle from "./PubMedArticle.vue";

export default defineComponent({
  name: "SearchResults",

  props: {
    count: {
      type: String,
      required: true,
    },
    results: {
      type: Array,
      required: true,
    },
    isPristine: {
      type: Boolean,
      required: true,
    },
  },

  components: {
    PubMedArticle,
  },
});
</script>

<style scoped>
li:not(:last-of-type) {
  margin-bottom: 2em;
}

section {
  background-color: rgb(223, 223, 222);
}
</style>
