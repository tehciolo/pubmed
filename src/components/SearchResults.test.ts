import { mount } from '@vue/test-utils'
import SearchResults from './SearchResults.vue'
import PubMedArticle from './PubMedArticle.vue'

test('mount component', async () => {
  expect(SearchResults).toBeTruthy()

  const wrapper = mount(SearchResults, {
    propsData: {
      count: '1337',
      results: [],
      isPristine: true
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
});

test('renders an article for every result item', async () => {
  expect(SearchResults).toBeTruthy()

  const wrapper = mount(SearchResults, {
    propsData: {
      count: '3',
      results: [
        { uid: 0, authors: [] },
        { uid: 1, authors: [] },
        { uid: 2, authors: [] },
      ],
      isPristine: false
    },
  })

  const articles = wrapper.findAllComponents(PubMedArticle);

  expect(articles.length).toBe(3)
});
