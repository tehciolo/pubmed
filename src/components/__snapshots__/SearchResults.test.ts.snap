// Vitest Snapshot v1

exports[`mount component 1`] = `
"<section data-v-1d54fe2e=\\"\\" class=\\"section is-medium\\">
  <p data-v-1d54fe2e=\\"\\"> PubMed is a free resource supporting the search and retrieval of biomedical and life sciences literature with the aim of improving health â€“ both globally and personally. Enter a search term above to start. </p>
</section>"
`;
